src/Kernel-Tests/ExceptionTest.class.st

```diff
@@ -320,12 +352,45 @@ ExceptionTest >> testSimpleResume [
 	self assertSuccess: (ExceptionTester new runTest: #simpleResumeTest ) 
 ]
 
+{ #category : #'tests - exceptiontester' }
+ExceptionTest >> testSimpleRetry [
+	self assertSuccess: (ExceptionTester new runTest: #simpleRetryTest ) 
+]
+
+{ #category : #'tests - exceptiontester' }
+ExceptionTest >> testSimpleRetryUsing [
+	self assertSuccess: (ExceptionTester new runTest: #simpleRetryUsingTest ) 
+]
+
 { #category : #'tests - exceptiontester' }
 ExceptionTest >> testSimpleReturn [
 	self assertSuccess: (ExceptionTester new runTest: #simpleReturnTest ) 
 ]
 
-{ #category : #tests }
+{ #category : #'tests - exceptiontester' }
 ExceptionTest >> testTimeoutWithZeroDuration [
 	self assertSuccess: (ExceptionTester new runTest: #simpleTimeoutWithZeroDurationTest ) 
 ]
+
+{ #category : #'tests - handling' }
+ExceptionTest >> testUnhandledErrorWhenHandlerPassesOriginalException [
+	| executed semaphore |
+	executed := false.
+	semaphore := Semaphore new.
+	[ [ 
+			[ 1/0 ] on: UnhandledError do: [ :e | executed := true ]
+		] on: ZeroDivide do: [:z | z pass]
+		. semaphore signal] fork.
+	semaphore wait: 2 seconds.
+	self assert: executed
+]
+
+{ #category : #'tests - handling' }
+ExceptionTest >> testUnhandledErrorWhenNoHandlers [
+	| executed semaphore |
+	executed := false.
+	semaphore := Semaphore new.
+	[[ 1/0 ] on: UnhandledError do: [ :e | executed := true ]. semaphore signal] fork.
+	semaphore wait: 2 seconds.
```

Comment:
```suggestion
	"we should not wait forever in case of wrong behaviour (general test timeout is disabled in #runCaseManaged)"
	semaphore wait: 2 seconds.
```